### PR TITLE
Update Perspective and Room to set up socket.io connection to separate

### PR DESCRIPTION
**Pull request by iamigo**
real-time application if separate real-time application is configured
via REALTIME_APPLICATION environment variable. Default to current
behavior (i.e. serve real-time events from the main Refocus application
if no real-time application is configured).

Also: only create redis subscribers in the main app if NO separate
real-time application is configured.

### CODE REVIEW

File: tests/realtime/pubSub.js [hunk 1]
```diff
@@ -67,6 +67,12 @@ describe('tests/realtime/pubsub.js >', () => {
     });
 
     it('subscribers receive all published messages', (done) => {
+      // No subscribers if running with separate real-time application
+      if (process.env.REALTIME_APPLICATION) {
```
Comment (jgraff2) [inline tests/realtime/pubSub.js]: Does this actually get set when running the tests?

File: cache/redisCache.js [hunk 2]
```diff
@@ -61,14 +61,22 @@ if (featureToggles.isFeatureEnabled('enableRedisConnectionLogging')) {
...
+  }
+
   pubPerspectives.push(redis.createClient(rp, opts));
 });
 
-const subBot = redis.createClient(rconf.instanceUrl.pubsubBots, opts);
-subBot.subscribe(rconf.botChannelName);
+// Only create subscribers here if we're doing real-time events from main app
+let subBot;
+if (!process.env.REALTIME_APPLICATION) {
```
Comment (jgraff2) [inline cache/redisCache.js]: Any reason for not using a toggle?

File: tests/realtime/pubSub.js (hunk 1 above)
Comment (iamigo) [reply tests/realtime/pubSub.js]: This new script in `package.json`:

```
    "test-realtime-application": "REALTIME_APPLICATION=https://abc.def npm run test-with-coverage -- tests/realtime/pubSub.js",
```

File: cache/redisCache.js (hunk 2 above)
Comment (iamigo) [reply cache/redisCache.js]: Updated to set up and use toggle @jgraff2

**Pull request merged**